packages/core/src/index.mts

```diff
@@ -181,41 +182,43 @@ export function usePagination(
   {
     active,
     pageSize = 7,
+    loop = true,
   }: {
     active: number;
     pageSize?: number;
+    loop?: boolean;
   },
 ) {
   const { rl } = context.getStore();
-  const state = useRef({
-    pointer: 0,
-    lastIndex: 0,
-  });
 
   const width = cliWidth({ defaultWidth: 80, output: rl.output });
   const lines = breakLines(output, width).split('\n');
 
-  // Make sure there's enough lines to paginate
-  if (lines.length <= pageSize) {
-    return output;
-  }
-
-  const middleOfList = Math.floor(pageSize / 2);
-
-  // Move the pointer only when the user go down and limit it to the middle of the list
-  const { pointer: prevPointer, lastIndex } = state.current;
-  if (prevPointer < middleOfList && lastIndex < active && active - lastIndex < pageSize) {
-    state.current.pointer = Math.min(middleOfList, prevPointer + active - lastIndex);
-  }
-
-  state.current.lastIndex = active;
-
-  // Duplicate the lines so it give an infinite list look
-  const infinite = [lines, lines, lines].flat();
-  const topIndex = Math.max(0, active + lines.length - state.current.pointer);
-
-  const section = infinite.splice(topIndex, pageSize).join('\n');
-  return section + '\n' + chalk.dim('(Move up and down to reveal more choices)');
+  const [lastActive, setLastActive] = useState(active);
+  useEffect(() => {
+    setLastActive(active);
+  }, [active, setLastActive]);
+
```

Comment:
Just trying to separate state and make it more idiomatic. Please let me know if there are any concerns.

Reply:
Setting state from an effect is often seen as an anti-pattern in React since it'll cause extra re-renders. This will also be true for Inquirer; and somewhat more so since frequent re-renders definitively can cause the prompt to flicker.

I didn't dig deep in the code yet; but if you need to keep a state between renders, I'd recommend `useRef` instead.

Reply:
I've move this back to refs, thanks!